app/models/katello/host/subscription_facet.rb

```diff
@@ -53,7 +57,6 @@ def import_database_attributes(consumer_params = candlepin_consumer.consumer_att
         self.update_installed_products(consumer_params['installedProducts']) if consumer_params.key?('installedProducts')
         self.purpose_role = consumer_params['role'] unless consumer_params['role'].nil?
         self.purpose_usage = consumer_params['usage'] unless consumer_params['usage'].nil?
-        self.purpose_addons = consumer_params['addOns'] unless consumer_params['addOns'].nil?
```

Comment:
I looked through the code and there are some instances where we update the subscription facet attributes not based on what was passed in (via UI update or client host checkin). In these situations we update the subscription facet based on the values in candlepin. Notice the signature of the method which does this:

```
def import_database_attributes(consumer_params = candlepin_consumer.consumer_attributes)
```

So, I think we need to restore this behavior that was removed. Since we no longer have purpose_addons as an attribute you can use the same approach as the host extensions with purpose_addon_ids

Reply:
Yes. That makes sense. I have updated the subscription_facet.rb and also have added the related test. Could you please review.

Reply:
Looks good. Can you add the assertion back which tests the functionality?